segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -0,0 +1,204 @@
+        b_cov = B_CENTROID_COV
+    else:
+        mean_b = distr_B['mean']
+        var_b = distr_B['cov']
+
+    if not distr_C:
+        c_mean = (height * C_CENTROID_FACTOR, width * C_CENTROID_FACTOR)
+        c_cov = C_CENTROID_COV
+    else:
+        mean_ac = distr_C['mean']
+        var_ac = distr_C['var']
+
+    # if specified, set the random seed
+    if seed:
+        np.random.seed(seed)
+
+    # use the multivariate_normal distribution to generate the points
+    a_points = np.random.multivariate_normal(a_mean, a_cov, num_A)
+    b_points = np.random.multivariate_normal(b_mean, b_cov, num_B)
+    c_points = np.random.multivariate_normal(c_mean, c_cov, num_C)
+
+    return a_points, b_points, c_points
+
+
+def point_init_dist_matrix(size_img=(1024, 1024), num_A=100, num_B=100, num_C=100, distr_A=None, distr_B=None, distr_C=None, seed=None):
+    """
+    This function generates random points using the get_random_centroid_centers function and from that
+    generates a distance matrix.
+
+    Each row and column of the matrix represents a specific cell, and the elements represent the distance
+    from the respective cell centroids.
+
+    The format of the matrix in terms of distances would look like:
+        A-A   A-B   A-C
+        B-A   B-B   B-C
+        C-A   C-B   C-C
+
+    Args:
+        size_img: a tuple indicating the size of the image. Default 1024 x 1024.
+        num_A: the number of A centroids to generate. Used by the get_random_centroid_centers function. Default 100.
+        num_B: similar to num_A
+        num_C: similar to num_C
+        distr_A: a dict indicating the mean and covariance matrix of the multivariate distribution we pull A centroids from.
+            Used by get_random_centroid_centers. Default None, and will use predefined parameters.
+        distr_B: dimilar to distr_A
+        distr_C: similar to distr_C
+        seed: whether to fix the random seed or not. Used by get_random_centroid_centers. Useful for testing.
+            Should be a specified integer value. Default None.
+    """
+
+    # generate points for cell types A, B, and C
+    a_points, b_points, c_points = get_random_centroid_centers(size_img, num_A, num_B, num_C, distr_A, distr_B, distr_C, seed)
+
+    # compute the distances between each point pair
+    a_a_dist = cdist(a_points, a_points)
+    a_b_dist = cdist(a_points, b_points)
+    a_c_dist = cdist(a_points, c_points)
+    b_b_dist = cdist(b_points, b_points)
+    b_c_dist = cdist(b_points, c_points)
+    c_c_dist = cdist(c_points, c_points)
+
+    # create each matrix row
+    # we need to correct aa, bb, and cc to ensure symmetry
+    first_row = np.concatenate(((a_a_dist + a_a_dist.T) / 2, a_b_dist, a_c_dist), axis=1)
+    second_row = np.concatenate((a_b_dist.T, (b_b_dist + b_b_dist.T) / 2, b_c_dist), axis=1)
+    third_row = np.concatenate((a_c_dist.T, b_c_dist.T, (c_c_dist + c_c_dist) / 2), axis=1)
```

Comment:
I'm pretty sure that `a_a_dist` is guaranteed to be symmetric.  no real harm in ensuring symmetry anyways, but for big matrices, this might incur a performance drop.  Consider this comment as negligible as the performance difference.